atomic_reactor/plugins/exit_koji_import.py

```diff
@@ -98,9 +98,15 @@ def get_output(self, worker_metadatas):
         :return: list, containing dicts of partial metadata
         """
         outputs = []
+        has_pulp_pull = PLUGIN_PULP_PULL_KEY in self.workflow.postbuild_results
         for platform in worker_metadatas:
             for instance in worker_metadatas[platform]['output']:
                 instance['buildroot_id'] = '{}-{}'.format(platform, instance['buildroot_id'])
+
+                # update image ID with pulp_pull results
+                if 'extra' in instance and platform == "x86_64" and has_pulp_pull:
```

Comment:
IIRC `koji_upload` expects platform name to be the architecture of the build, so its safe to rely on platform (although indeed feels bad). Using goarch for that would bump the required osbs-client version